### ISSUE TITLE
Visual Studio Code is broken after an update (again)

### ISSUE THREAD

**Issue by admercs** [NixOS/nixpkgs#241337]
### Describe the bug
VS Code (and VS Codium) opens with a blank white screen after installation. An Electron issue?

### Steps To Reproduce
nixos-rebuild switch --upgrade

### Expected behavior
I expect the app to open to the welcome screen.

### Additional context
This is a long-standing issue: https://github.com/NixOS/nixpkgs/issues/56236

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

**Comment by Majiir** [comment]
Works for me on latest 23.05 (0de86059128947b2438995450f2c2ca08cc783d5, same commit as your system). Can you share more details? Are you using Wayland or X11? What is your config for enabling VS Code?

**Comment by teto** [comment]
looks like a wayland issue have you exported NIXOS_OZONE... etc ? if you are hit by this, you could write a test for it

**Comment by admercs** [comment]
@teto Indeed, it appears to have been a Wayland issue! Setting the following sessionVariable in my configuration resolved it:

`environment.sessionVariables.NIXOS_OZONE_WL = "1";`

However, the window on Electron applications is in light mode while the rest of my computer is set to dark mode. How do I change the Electron app theme?

**Comment by SimonBrandner** [comment]
Hey, not sure what exactly is happening on your side but https://github.com/NixOS/nixpkgs/issues/242113#issuecomment-1626299441 might be worth noting

**Comment by eclairevoyant** [comment]
This is pending vscode to move to electron 28 or higher, see electron/electron#40366.
